ipv8/test/mocking/rest/peer_communication.py

```diff
@@ -0,0 +1,113 @@
+
+from abc import abstractmethod
+
+
+class PostStyleRequests(object):
```

Comment:
Certainly. I initially added this class, with the purpose of it functioning like an interface. When I initially wrote it, there was only an `attestation_endpoint` (now there are multiple endpoints), and this class defined an interface for all possible `POST` type requests then. I thought it would be a good idea to add it since it would separate the interface from the implementation, that is to say, there may be another protocol, other than HTTP, which may implement these methods, and still be RESTful, and this protocol may be implemented in IPv8 later on in the future. This interface would be used in asserts, to ensure that the passed object does have these methods (by being an instance of `PostStyleRequests`). The same applies for `GetStyleRequests`, but for `GET` requests. If need be, they can both be removed.

Reply:
Should I remove them? Moreover, if I should remove them, should I then merge `HTTPGetRequester` and `HTTPPostRequester` (i.e. the classes which implement these 2 interfaces) together into a big class which implements  request methods, or should I leave them separated?